backend/multimodalexplorer/api/endpoints/search.py

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Any, Dict
+
+from fastapi import APIRouter, Request
+from fastapi.responses import JSONResponse
+
+from multimodalexplorer.functions.search_faiss_index import SearchFaissIndex
+from multimodalexplorer.utils.utils import select_params
+
+
+def create_search_router(args: Dict[str, Any]) -> APIRouter:
+    router = APIRouter()
+
+    @router.post("/search_data")
+    async def get_search_result(request: Request) -> Dict[str, Any]:
+
+        data = await request.json()
```

Comment:
the benefit of fastapi is that it will deal with param typing and checks for you, see https://fastapi.tiangolo.com/tutorial/body/#import-pydantics-basemodel

so you can just declare a "model" for your data and let fastapi check that the schema is respected by the query